src/app/core/services/prices/prices.service.ts

```diff
@@ -34,11 +41,15 @@ export class PricesService {
           params = params.set('customerID', customer.customerNo);
         }
       }),
-      switchMap(() =>
-        this.apiService
+      withLatestFrom(this.store.pipe(select(getCurrentBasket))),
+      switchMap(([, basket]) => {
+        if (basket?.commonShipToAddress) {
+          params = params.append('shipToAddress', basket.commonShipToAddress.urn);
```

Comment:
1. Why is the method append() used? Is it possible, that the 'shipToAddress' parameter have multiple values?
2. Optional: Maybe it is a good idea to restructure the pipe. I would like, that the parameters are all set within one operator. Another one is just responsible to fetch the new product prices with the given parameters.